packages/lib-core/src/extensions/actions.ts

```diff
@@ -0,0 +1,119 @@
+import type { ReactNode } from 'react';
+import type { ExtensionHook, ExtensionK8sKindVersionModel } from '../api/common-types';
+import type { CodeRef, Extension } from '../types/extension';
+import type { AccessReviewResourceAttributes } from './core-types';
+
+/** ActionProvider contributes a hook that returns list of actions for specific context */
+export type ActionProvider = Extension<
+  'core.action/provider',
+  {
+    /** The context ID helps to narrow the scope of contributed actions to a particular area of the application. Ex - topology, helm */
+    contextId: string | 'resource';
```

Comment:
TS coerces this to just `string` so I'm not sure why we need the `| 'resource'` part.

Reply:
Yeah, this felt a bit funky when I moved it over. But I thought it had some purpose when type suggestion. Removing the `| 'resource'` part!